### PR TITLE
Show welcome modal to new signups only

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8132

## Description of Changes
Updated to only show welcome modal for new user signups.

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_USER_ONBOARDING_ENABLED`
- Verify that you can signup with multiple auth options
- Verify you see the welcome modal, and you keep seeing it after page refresh, until you set username in the welcome modal.

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

**Review by masvelio — APPROVED** [review]
lgtm

File: packages/commonwealth/server/migrations/20240612153431-added-isWelcomeOnboardFlowComplete-to-users.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    return queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.addColumn(
+        'Users',
+        'isWelcomeOnboardFlowComplete',
```
Comment (kurtisassad) [inline packages/commonwealth/server/migrations/20240612153431-added-isWelcomeOnboardFlowComplete-to-users.js:9]: We should be using snake_case for all field names. SQL standard is case-insensitive which is why we need the quotation marks when we use mixed casing which is a hassle.

**Comment by kurtisassad** [comment]
Curious why we didn't go with a localStorage approach here on the front end. Seems like extra work storing this flag on the backend that will be true for most users, when a new user would be expected to perform the welcome modal stuff on first login.

**Comment by masvelio** [comment]
@kurtisassad if user logs in with different browser/device or clears cache, then he would get the welcome modal again, despite he already completed it

**Comment by mzparacha** [comment]
> Curious why we didn't go with a localStorage approach here on the front end. Seems like extra work storing this flag on the backend that will be true for most users, when a new user would be expected to perform the welcome modal stuff on first login.

@kurtisassad, because of the multiple devices sync condition that @masvelio mentioned (Thanks @masvelio).

Another reason is that we also depend on a valid username for a user to be considered onboarded and atm a lot of users don't have a valid name (it's either null or Anonymous -- both considered invalid in the new onboarding flow). If we go with the local storage approach, these existing users will also see the welcome modal.

Add a source of truth to the db, fixes both these issues.

**Review by kurtisassad — APPROVED** [review]

**Review by kurtisassad — APPROVED** [review]

**Pull request merged**